File: src/cpp/FizzBuzz2.c
#include<stdio.h> 
// defineでマクロ関数を定義
#define IS_FIZZ(i) (i % 3 == 0)
#define IS_BUZZ(i) (i % 5 == 0)

char buf[256];

// FizzBuzzを返す関数を定義 文字列の配列を返す　fizz-buzzという関数名
char* fizzbuzz(int i) {
  // 関数を使って条件を次々と判定し、一番最初に該当したif文のみreturn実行
  if (IS_FIZZ(i) && IS_BUZZ(i)) return "FizzBuzz";
  if (IS_FIZZ(i)) return "Fizz";
  if (IS_BUZZ(i)) return "Buzz";
  sprintf(buf, "%d", i); // iを%dで指定したフォーマットでbufに保存する
  return buf; // fizz buzz番号も返す
}

// メイン関数
int main(void) {
  // 繰り返しfizz_buzz関数を呼び出す
  int i;
  for(i = 1; i <= 100; ++i) {
    printf("%s\n", fizzbuzz(i)); // %sで文字配列指定　\nで改行
  }
  return 0;
}

File: src/FizzBuzz Variety Sample Codes/fizzbuzz2.go
package main
import ("fmt")

// メイン関数
func main() {
    // 構造体を生成して初期化 --- (*1)
    fb := FizzBuzz{1, 100}
    fb.Run() // 実行
}

// FizzBuzz構造体を定義 --- (*2)
type FizzBuzz struct {
    Cur int
    Max int
}
// 構造体FizzBuzzで処理実行のメソッド --- (*3)
func (p *FizzBuzz) Run() {
    // 指定回数だけ繰り返す --- (*4)
    for ; p.Cur <= p.Max; p.Cur++ {
        fmt.Println(p.GetValue())
    }
}
// 構造体FizzBuzzで条件を判定するメソッド --- (*5)
func (p* FizzBuzz) IsFizz() bool {
    return p.Cur % 3 == 0
}
func (p* FizzBuzz) IsBuzz() bool {
    return p.Cur % 5 == 0
}
// 構造体FizzBuzzでFizzBuzzの値を取得するメソッド --- (*6)
func (p *FizzBuzz) GetValue() string {
    switch {
    case p.IsFizz() && p.IsBuzz():
        return "FizzBuzz"
    case p.IsFizz():
        return "Fizz"
    case p.IsBuzz():
        return "Buzz"
    default:
        return fmt.Sprint(p.Cur)
    }
}
